src/reader/mod.rs

```diff
@@ -759,6 +818,60 @@ impl<R> Reader<R> {
     pub const fn decoder(&self) -> Decoder {
         self.state.decoder()
     }
+
+    /// Get the direct access to the underlying reader, but tracks the amount of
+    /// read data and update [`Reader::buffer_position()`] accordingly.
+    ///
+    /// # Example
+    ///
+    /// This example demonstrates, how it is possible to read embedded binary data.
+    /// Such XML documents are exist in the wild.
+    ///
+    /// ```
+    /// # use pretty_assertions::assert_eq;
+    /// use std::io::{BufRead, Read};
+    /// use quick_xml::events::{BytesEnd, BytesStart, Event};
+    /// use quick_xml::reader::Reader;
+    ///
+    /// let mut reader = Reader::from_str("<tag>binary << data&></tag>");
+    /// //                                 ^    ^               ^     ^
+    /// //                                 0    5              21    27
+    ///
+    /// assert_eq!(
+    ///     (reader.read_event().unwrap(), reader.buffer_position()),
+    ///     // 5 - end of the `<tag>`
+    ///     (Event::Start(BytesStart::new("tag")), 5)
+    /// );
+    ///
+    /// // Reading directly from underlying reader will not update position
+    /// // let mut inner = reader.get_mut();
+    ///
+    /// // Reading from the stream() advances position
+    /// let mut inner = reader.stream();
+    ///
+    /// // Read binary data. We somehow should known its size
```

Comment:
We must know its size

Reply:
Also somewhere (maybe up above, not here), we should mention the two conditions that the user must uphold when using this API and the potential consequences if it's misused.

Reply:
Mmm... what two conditions? I added a note that read data will not be returned in subsequent `Event`s

Reply:
> the user should manually:
> 
> -  ensure, that it is reading text (binary data)
> -  stop reading when text (binary data) ends
>

Reply:
@dralley, what do you think about my last comment? If you think, that we can write something clear do not hesitate to write it -- I decidedly have no thoughts. I mean, that just "when binary data ends" is not clear enough, because this literally can be any condition which is already clear.